Create partition timeout if there are many partitions existed in collection

**Please state your issue using the following template and, most importantly, in English.**

**Describe the bug**
```
__________________ TestCreateBase.test_create_partition_limit __________________

self = <test_partition.TestCreateBase object at 0x7f265d06d8d0>
connect = <milvus.client.stub.Milvus object at 0x7f264c156f28>
collection = 'partition_AE8VNnL3'
args = {'handler': 'GRPC', 'ip': 'milvus-ci-master-1153-single-centos7-gpu.milvus.svc.cluster.local', 'port': 19530, 'service_name': 'milvus-ci-master-1153-single-centos7-gpu'}

    @pytest.mark.level(2)
    def test_create_partition_limit(self, connect, collection, args):
        '''
        target: test create partitions, check status returned
        method: call function: create_partition for 4097 times
        expected: exception raised
        '''
        threads_num = 16
        threads = []
        if args["handler"] == "HTTP":
            pytest.skip("skip in http mode")
    
        def create(connect, threads_num):
            for i in range(4096 // threads_num):
                tag_tmp = gen_unique_str()
                connect.create_partition(collection, tag_tmp)
    
        for i in range(threads_num):
            m = get_milvus(host=args["ip"], port=args["port"], handler=args["handler"])
            t = threading.Thread(target=create, args=(m, threads_num, ))
            threads.append(t)
            t.start()
        for t in threads:
>           t.join()

test_partition.py:65: 
```
```
  File "/usr/local/lib/python3.6/threading.py", line 884, in _bootstrap
    self._bootstrap_inner()
  File "/usr/local/lib/python3.6/threading.py", line 916, in _bootstrap_inner
    self.run()
  File "/usr/local/lib/python3.6/threading.py", line 864, in run
    self._target(*self._args, **self._kwargs)
  File "/home/jenkins/agent/workspace/milvus-ci_master/tests/milvus_python_test/test_partition.py", line 57, in create
    connect.create_partition(collection, tag_tmp)
  File "/usr/local/lib/python3.6/site-packages/milvus/client/stub.py", line 35, in inner
    return func(self, *args, **kwargs)
  File "/usr/local/lib/python3.6/site-packages/milvus/client/stub.py", line 514, in create_partition
    return handler.create_partition(collection_name, partition_tag, timeout)
  File "/usr/local/lib/python3.6/site-packages/milvus/client/grpc_handler.py", line 42, in handler
    return func(self, *args, **kwargs)
  File "/usr/local/lib/python3.6/site-packages/milvus/client/grpc_handler.py", line 487, in create_partition
    response = rf.result()
  File "/usr/local/lib/python3.6/site-packages/grpc/_channel.py", line 616, in result
    timeout=timeout)
  File "/usr/local/lib/python3.6/site-packages/grpc/_common.py", line 138, in wait
    _wait_once(wait_fn, MAXIMUM_WAIT_TIMEOUT, spin_cb)
  File "/usr/local/lib/python3.6/site-packages/grpc/_common.py", line 103, in _wait_once
    wait_fn(timeout=timeout)
  File "/usr/local/lib/python3.6/threading.py", line 299, in wait
    gotit = waiter.acquire(True, timeout)
```
**Steps/Code to reproduce behavior**


**Expected behavior**
A clear and concise description of what you expected to happen.

**Method of installation**
- [x] Docker/cpu
- [ ] Docker/gpu
- [ ] Build from source

**Environment details**
- Hardware/Softwars conditions (OS, CPU, GPU, Memory)


- Milvus version (master or released version)
master-branch

**Configuration file**
Settings you made in `server_config.yaml` or `milvus.yaml`

```yaml
paste-file-content-here
```

**Screenshots**
If applicable, add screenshots to help explain your problem.

**Additional context**
Add any other context about the problem here.

```
self = <test_partition.TestCreateBase object at 0x7f2479e33a90>
connect = <milvus.client.stub.Milvus object at 0x7f2479dc9c88>
collection = 'partition_D6JrVZ3o'
args = {'handler': 'GRPC', 'ip': 'milvus-ci-0-11-0-38-single-centos7-gpu.milvus.svc.cluster.local', 'port': 19530, 'service_name': 'milvus-ci-0-11-0-38-single-centos7-gpu'}

    @pytest.mark.level(2)
    def test_create_partition_limit(self, connect, collection, args):
        '''
        target: test create partitions, check status returned
        method: call function: create_partition for 4097 times
        expected: exception raised
        '''
        threads_num = 16
        threads = []
        if args["handler"] == "HTTP":
            pytest.skip("skip in http mode")
    
        def create(connect, threads_num):
            for i in range(4096 // threads_num):
                tag_tmp = gen_unique_str()
                connect.create_partition(collection, tag_tmp)
    
        for i in range(threads_num):
            m = get_milvus(host=args["ip"], port=args["port"], handler=args["handler"])
            t = threading.Thread(target=create, args=(m, threads_num, ))
            threads.append(t)
            t.start()
        for t in threads:
>           t.join()

test_partition.py:65: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
/usr/local/lib/python3.6/threading.py:1056: in join
    self._wait_for_tstate_lock()
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

self = <Thread(Thread-8647, started 139792016713472)>, block = True
timeout = -1

    def _wait_for_tstate_lock(self, block=True, timeout=-1):
        # Issue #18808: wait for the thread state to be gone.
        # At the end of the thread's life, after all knowledge of the thread
        # is removed from C data structures, C code releases our _tstate_lock.
        # This method passes its arguments to _tstate_lock.acquire().
        # If the lock is acquired, the C code is done, and self._stop() is
        # called.  That sets ._is_stopped to True, and ._tstate_lock to None.
        lock = self._tstate_lock
        if lock is None:  # already determined that the C code is done
            assert self._is_stopped
>       elif lock.acquire(block, timeout):
E       Failed: Timeout >360.0s

/usr/local/lib/python3.6/threading.py:1072: Failed
```

--- a/tests/milvus_python_test/entity/test_insert.py
+++ b/tests/milvus_python_test/entity/test_insert.py
@@ -148,6 +148,16 @@ class TestInsertBase:
         logging.getLogger().debug(res)
         assert res
 
+    def test_insert_segment_row_count(self, connect, collection):
+        nb = segment_row_count + 1
+        res_ids = connect.insert(collection, gen_entities(nb))
+        connect.flush([collection])
+        assert len(res_ids) == nb
+        stats = connect.get_collection_stats(collection)
+        assert len(stats['partitions'][0]['segments']) == 2
+        for segment in stats['partitions'][0]['segments']:
+            assert segment['row_count'] in [segment_row_count, 1]
+
     @pytest.fixture(
         scope="function",
         params=[
@@ -522,7 +532,7 @@ class TestInsertBase:
         method: disable autoflush and insert, get entity
         expected: the count is equal to 0
         '''
-        delete_nums = 500;
+        delete_nums = 500
         disable_flush(connect)
         ids = connect.insert(collection, entities)
         res = connect.get_entity_by_id(collection, ids[:delete_nums])

--- a/tests/milvus_python_test/entity/test_search.py
+++ b/tests/milvus_python_test/entity/test_search.py
@@ -1607,7 +1607,7 @@ class TestSearchInvalid(object):
         index_type = "BIN_IVF_FLAT"
         int_vectors, entities, ids = init_binary_data(connect, binary_collection)
         query_int_vectors, query_entities, tmp_ids = init_binary_data(connect, binary_collection, nb=1, insert=False)
-        connect.create_index(binary_collection, binary_field_name, {"index_type": index_type, "metric_type": "JACCARD", "params": {"nlist": 1024}})
+        connect.create_index(binary_collection, binary_field_name, {"index_type": index_type, "metric_type": "JACCARD", "params": {"nlist": 128}})
         query, vecs = gen_query_vectors(binary_field_name, query_entities, top_k, nq, search_params={"nprobe": 0}, metric_type="JACCARD")
         with pytest.raises(Exception) as e:
             res = connect.search(binary_collection, query)

--- a/tests/milvus_python_test/stability/test_restart.py
+++ b/tests/milvus_python_test/stability/test_restart.py
@@ -219,7 +219,7 @@ class TestRestartBase:
         delete_length = 1000
         loop = 10
         for i in range(loop):
-            delete_ids = ids[i*loop:i*loop+delete_length]
+            delete_ids = ids[i*delete_length:(i+1)*delete_length]
             delete_res = connect.delete_entity_by_id(collection, delete_ids)
         connect.flush([collection])
         connect.compact(collection, _async=True)

--- a/tests/milvus_python_test/test_partition.py
+++ b/tests/milvus_python_test/test_partition.py
@@ -39,7 +39,7 @@ class TestCreateBase:
     # TODO: enable
     @pytest.mark.level(2)
     @pytest.mark.timeout(1200)
-    def _test_create_partition_limit(self, connect, collection, args):
+    def test_create_partition_limit(self, connect, collection, args):
         '''
         target: test create partitions, check status returned
         method: call function: create_partition for 4097 times

